src/commands/SetPowerLevelCommand.ts

```diff
@@ -23,12 +23,39 @@ export async function execSetPowerLevelCommand(roomId: string, event: any, mjoln
     const level = Math.round(Number(parts[3]));
     const inRoom = parts[4];
 
+    const mjolnirId = await mjolnir.client.getUserId();
+
     let targetRooms = inRoom
         ? [await mjolnir.client.resolveRoom(inRoom)]
         : mjolnir.protectedRoomsTracker.getProtectedRooms();
 
+    let force = false;
+    if (parts[parts.length - 1] === "--force") {
+        force = true;
+        parts.pop();
+    }
+
     for (const targetRoomId of targetRooms) {
         try {
+            if (!force) {
+                if (target === mjolnirId || mjolnir.moderators.checkMembership(target)) {
+                    // don't let the bot demote itself or members of moderation room
+                    const currentLevels = await mjolnir.client.getRoomStateEvent(
+                        targetRoomId,
+                        "m.room.power_levels",
+                        "",
+                    );
+                    const targetLevel = currentLevels["users"][mjolnirId];
+                    if (level < targetLevel) {
```

Comment:
`targetLevel` at a glance feels like the destination level, making this statement a bit confusing. Maybe `currentLevel` would fit better?